contribs/gmf/src/directives/routing.js

```diff
@@ -0,0 +1,213 @@
+goog.provide('gmf.routingComponent');
+
+goog.require('gmf');
+goog.require('gmf.RoutingService');
+
+
+gmf.module.value('gmfRoutingTemplateUrl',
+  /**
+   * @param {!angular.JQLite} $element Element.
+   * @param {!angular.Attributes} $attrs Attributes.
+   * @return {string} Template URL.
+   */
+  ($element, $attrs) => {
+    const templateUrl = $attrs['gmfRoutingTemplateUrl'];
+    return templateUrl !== undefined ? templateUrl :
+      `${gmf.baseTemplateUrl}/routing.html`;
+  }
+);
+
+
+/**
+ * @param {!angular.JQLite} $element Element.
+ * @param {!angular.Attributes} $attrs Attributes.
+ * @param {!function(!angular.JQLite, !angular.Attributes): string} gmfRoutingTemplateUrl Template function.
+ * @return {string} Template URL.
+ * @ngInject
+ */
+function gmfRoutingTemplateUrl($element, $attrs, gmfRoutingTemplateUrl) {
+  return gmfRoutingTemplateUrl($element, $attrs);
+}
+
+
+/**
+ * Component to provide OSRM routing.
+ *
+ * Example:
+ *
+ *  <gmf-routing
+ *    gmf-routing-active="ctrl.routingActive"
+ *    gmf-routing-map="::ctrl.map">
+ *  </gmf-routing>
+ *
+ * @htmlAttribute {boolean} gmf-routing-active Whether the component is
+ *     active or not.
+ * @htmlAttribute {ol.Map} gmf-routing-map The map.
+ * @ngdoc component
+ * @ngname gmfObjectediting
+ */
+gmf.routingComponent = {
+  controller: 'GmfRoutingController as routeCtrl',
+  bindings: {
+    'active': '=gmfRoutingActive',
+    'map_': '<gmfRoutingMap'
+  },
+  templateUrl: gmfRoutingTemplateUrl
+};
+
+gmf.module.component('gmfRouting', gmf.routingComponent);
+
+
+/**
+ * The controller for the routing directive.
+ * @param {!angular.Scope} $scope Scope.
+ * @param {!gmf.RoutingService} gmfRoutingService service for OSRM routing
+ * @param {!angular.$q} $q Angular q service
+ * @constructor
+ * @private
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfRoutingController
+ */
+gmf.GmfRoutingController = function($scope, gmfRoutingService, $q) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.$scope_ = $scope;
+
+  /**
+   * @type {gmf.RoutingService}
+   * @private
+   */
+  this.gmfRoutingService_ = gmfRoutingService;
+
+  /**
+   * @type {angular.$q}
+   * @private
+   */
+  this.$q_ = $q;
+
+  /**
+   * @type {ol.Map}
+   * @private
+   */
+  this.map_;
+
+  /**
+   * @type {ol.Feature}
+   * @private
+   */
+  this.startFeature_ = null;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.startFeatureLabel_ = '';
```

Comment:
This variable is used in the template, it should be rename to `this.startFeatureLabel` and marked as `@export` instead of `@private`